swarm/src/lib.rs

```diff
@@ -249,6 +249,16 @@ pub enum SwarmEvent<TBehaviourOutEvent, THandlerErr> {
     Dialing(PeerId),
 }
 
+impl<TBehaviourOutEvent, THandlerErr> SwarmEvent<TBehaviourOutEvent, THandlerErr> {
+    /// Extract the `TBehaviourOutEvent` from this [`SwarmEvent`] in case it is the `Behaviour` variant, otherwise fail.
+    pub fn try_into_behaviour_event(self) -> Result<TBehaviourOutEvent, Self> {
```

Comment:
Is there a case where a user would call this method and not unwrap the result? If not, what do you think about panicking here instead of returning an Err?

Reply:
Given that this is a function in production code, I don't think panicking is a good option here.

Reply:
Woops, I overlooked that it is in `swarm` and not `swarm-test`. Agree with you.